worker-markup/src/main/java/com/hpe/caf/worker/markup/MarkupHeadersAndBody.java

```diff
@@ -154,6 +170,56 @@ else if (line.contains(": ")) {
     }
 
     /**
+     * Handle the correct markup of header fields which are surrounded with '*', i.e *Sent:* xxx.
+     *
+     * @param emailText the email text that is to be marked up.
+     * @param lines the emailText as separate lines.
+     * @param nattyParser the natty parser to detect the date language.
+     * @param headersElement the header element to add headers to.
+     * @return the email body index.
+     */
+    private int handleHeaderWithAsterisks(final String emailText, final String[] lines, final Parser nattyParser,
+                                          final Element headersElement, int bodyIndex)
+    {
+        // Check to see if the the from field is split onto two lines.
+        final Matcher splitFromFieldMatcher = fromFieldSplitOntoTwoLines.matcher(emailText);
+        final List<String> asteriskHeadersAlreadyMarkedUp = new ArrayList<>();
+
+        if (splitFromFieldMatcher.find()) {
+            for (final String line : lines) {
+                // For a line which matches the format "*From:*example.emailaddress.com",
+                // concat this and the second part of the 'From' field "[mailto:example.emailaddress.com] *On Behalf Of *Bloggs, Joe"
+                // So that the whole value is marked up with <From>
+                if (line.equals(splitFromFieldMatcher.group("FirstPartFromField"))) {
+                    asteriskHeadersAlreadyMarkedUp.add(line);
+                    bodyIndex++;
+                    final String fullFromFieldValue = line + "\n" + splitFromFieldMatcher.group("SecondPartFromField");
```

Comment:
Updated the regex to only match if there is a new line.

This wasn't breaking the unit test because:
If the full `From` value was on one line, it would match the regex and enter the for loop.

However, since the current line was the full `From` value it would never match the `if` or `else if` statement inside the for loop. 

Essentially, it was matching the regex and entering the for loop, but it wasn't doing anything.